### ISSUE TITLE
JSON validation duplicate name problem

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#6720]
Issue overview
--------------
User file with two material objects named "Standard insulation_0.1" and "Standard insulation_0.01" results in
```
   ** Severe  ** Duplicate name found. name: "Standard insulation_0.01". Overwriting existing object.
```
Changing one of them to "Standard insulation_0.01a" fixes the problem.

### Details
Some additional details for this issue (if relevant):
 - Platform Win 10
 - Version of EnergyPlus 8.9.0
 - Helpdesk ticket number 12634

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added EnergyPlusDevSupport\DefectFiles
 - [x] Ticket added to Pivotal for defect (development team task)
 - [x] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by eringold** [comment]
I just ran into the same error (v8.9.0) from a BuildingSurface:Detailed, which I verified did not have a duplicate name with anything else in the idf:
```
** Severe  ** Duplicate name found. name: "01_BIOCHEM1 - Wall 0:a". Overwriting existing object.
```
The input file was made with OpenStudio, which as I understand would not allow a duplicate name for that object to begin with. Changing the name to "01_BIOCHEM1 - 0aWall" fixed the problem (as well as for ~14 other surfaces with similarly structured names that threw an error).

**Comment by jmarrec** [comment]
Fixed in #7036
